DMX.Core.Api/Services/Orchestrations/LabOrchestrationService.Validations.cs

```diff
@@ -0,0 +1,74 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// ---------------------------------------------------------------
+
+using System;
+using DMX.Core.Api.Models.Foundations.Labs;
+using DMX.Core.Api.Models.Foundations.Labs.Exceptions;
+
+namespace DMX.Core.Api.Services.Orchestrations
+{
+    public partial class LabOrchestrationService
+    {
+        private void ValidateLabOnAdd(Lab lab)
+        {
+            ValidateLabIsNotNull(lab);
+
+            Validate(
+                (Rule: IsInvalid(lab.Id), Parameter: nameof(Lab.Id)),
+                (Rule: IsInvalidId(lab.ExternalId), Parameter: nameof(Lab.ExternalId)),
+                (Rule: IsInvalid(lab.Name), Parameter: nameof(Lab.Name)),
+                (Rule: IsInvalid(lab.Description), Parameter: nameof(Lab.Description)),
+                (Rule: IsInvalid(lab.Status), Parameter: nameof(Lab.Status)));
```

Comment:
You don't need any of these validations.
Every service will only validate what it needs - what that basically means if your service is only just adding an object (unless it's a foundation service) you should validate here if the lab is null or not.

Reply:
Is that because we don't actually use any of these fields in the logic of the add method?
And the foundation service add method doesn't either, but by virtue of it being a foundation service we need to validate everything anyway?